tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -165,5 +167,84 @@ public void AddResponseHeadersToSpanAndCleanup()
         _httpTransport.DisposeAdditiveContext();
     }
 
+    internal static Dictionary<string, object> ExtractCookiesFromRequest(HttpRequest request)
+    {
+        var cookies = RequestDataHelper.GetCookies(request);
+        var cookiesDic = new Dictionary<string, object>();
+
+        if (cookies != null)
+        {
+            for (var i = 0; i < cookies.Count; i++)
+            {
+#if NETCOREAPP || NETSTANDARD
+                var cookie = cookies.ElementAt(i);
+                var keyForDictionary = cookie.Key ?? string.Empty;
+#else
+                var cookie = cookies[i];
+                var keyForDictionary = cookie.Name ?? string.Empty;
+#endif
+                if (!cookiesDic.TryGetValue(keyForDictionary, out var value))
+                {
+                    cookiesDic.Add(keyForDictionary, cookie.Value ?? string.Empty);
+                }
+                else
+                {
+                    if (value is string stringValue)
+                    {
+                        cookiesDic[keyForDictionary] = new List<string> { stringValue, cookie.Value ?? string.Empty };
+                    }
+                    else if (value is List<string> valueList)
+                    {
+                        valueList.Add(cookie.Value ?? string.Empty);
+                    }
+                    else
+                    {
+                        Log.Warning("Cookie {Key} couldn't be added as argument to the waf", keyForDictionary);
+                    }
+                }
+            }
+        }
+
+        return cookiesDic;
+    }
+
+#if NETFRAMEWORK
+    internal static Dictionary<string, object> ExtractHeadersFromRequest(NameValueCollection headers)
```

Comment:
I kinda prefered having no if/else in the generic `SecurityCoordinator` class and using the `SecurityCoordinator.Core` / `Security.Framework` to pivot between the net / core types

Reply:
I moved to securityCoordinator to avoid code duplication and to make sure that the behavior is the same in both cases even if that means that we have to add some preprocessor statements, but it could be moved back where it was. WDYT?

Reply:
maybe we could have helpers functions in the respective coordinators for these 2 lines of codes that differ ? and keep the main method `ExtractCookiesFromRequest` in `SecurityCoordinator` ?

Reply:
Sounds good! I have moved these lines to coordinators while keeping the main method in securityCoordinator